test/pypi-integration.py

```diff
@@ -92,31 +93,25 @@ def test_fail_to_upload_if_input_name_different_from_package_name(self):
                 }
             )
 
-    def test_upload_when_different_names_allowed(self):
-        rc = subprocess.run(['python', 'setup.py', 'sdist'], check=True, cwd=REPODIR)
-        print("sdist returned", rc)
-        with self.assertRaises(out.NamesValidationError) as context:
-            out.out(
-                os.path.join(REPODIR, 'dist'),
-                {
-                    'source': {'test': True, 'name': 'mismatching-name', 'name_must_match': False},
-                    'params': {'glob': '*.tar.gz'}
-                }
-            )
-
     def test_fail_to_upload_if_package_version_not_pep440_compliant(self):
```

Comment:
This tests was always broken since I introduced it 2.5 years ago :( 
which also means that no one ran the integration tests before merging my PR upstream :(